matminer/featurizers/site.py

```diff
@@ -1677,3 +1708,109 @@ def citations(self):
 
     def implementors(self):
         return ["Maxwell Dylla"]
+
+
+# TODO: Figure out whether to take NN-counting method as an option (see VoronoiFingerprint)
+class LocalPropertyDifference(BaseFeaturizer):
+    """Compute the different in elemental properties between site
+    and its neighboring sites.
+
+    Uses the Voronoi tessellation of the structure to determine the
+    neighbors of the site, and assigns each neighbor (:math:`n`) a
+    weight (:math:`A_n`) that corresponds to the area of the facet
+    on the tessellation corresponding to that neighbor.
+    The local property difference is then computed by
+    :math:`\frac{\sum_n {A_n |p_n - p_0|}}{\sum_n {A_n}}`
+    where :math:`p_n` is the property (e.g., atomic number) of a neighbor
+    and :math:`p_0` is the property of a site.
+
+    Features:
+        - "local property difference in [property]" - Weighted average
+            of differences between an elemental property of a site and
+            that of each of its neighbors, weighted by size of face on
+            Voronoi tessellation
+
+    References:
+         `Ward et al. _PRB_ 2017 <http://link.aps.org/doi/10.1103/PhysRevB.96.014107>`_
+    """
+
+    def __init__(self, data_source=MagpieData(), weight='area',
+                 properties=('Electronegativity')):
+        """ Initialize the featurizer
+
+        Args:
+            data_source (AbstractData) - Class from which to retrieve
+                elemental properties
+            weight (str) - What aspect of each voronoi facet to use to
+                weigh each neighbor (see VoronoiNN)
+            properties ([str]) - List of properties to use (default=
+        """
+        self.data_source = data_source
+        self.properties = properties
+        self.weight = weight
+
+    @staticmethod
+    def from_preset(preset):
+        """
+        Create a new LocalPropertyDifference class according to a preset
+
+        Args:
+            preset (str) - Name of preset
+        """
+
+        if preset == "ward-prb-2017":
+            return LocalPropertyDifference(
+                data_source=MagpieData(),
+                properties=["Number", "MendeleevNumber", "AtomicWeight",
+                            "MeltingT", "Column", "Row", "CovalentRadius",
+                            "Electronegativity", "NsValence", "NpValence",
+                            "NdValence", "NfValence", "NValance", "NsUnfilled",
+                            "NpUnfilled", "NdUnfilled", "NfUnfilled",
+                            "NUnfilled", "GSvolume_pa", "GSbandgap",
+                            "GSmagmom", "SpaceGroupNumber"]
+            )
+        else:
```

Comment:
technically you don't need the "else" here, if you haven't "returned" by now it's time for an error

Hard to say what's the more "proper" code here. Leaving the else "in" might be clearer than removing it? up to you

Reply:
I could go either way. I'll keep the 'else' in, because - to me - it makes it clearer that this is a case/select block.

Reply:
ok